processing/src/main/java/io/druid/query/groupby/epinephelinae/GroupByRowProcessor.java

```diff
@@ -66,18 +69,28 @@
       final GroupByQueryResource resource,
       final ObjectMapper spillMapper,
       final String processingTmpDir,
-      final int mergeBufferSize
+      final int mergeBufferSize,
+      final boolean wasQueryPushedDown
   )
   {
-    final GroupByQuery query = (GroupByQuery) queryParam;
+    GroupByQuery queryWithRenamedDimensions = null;
+    if (wasQueryPushedDown) {
+      List<DimensionSpec> newDims = new ArrayList<>(((GroupByQuery) queryParam).getDimensions().size());
+      for (DimensionSpec ds : ((GroupByQuery) queryParam).getDimensions()) {
+        DimensionSpec dim = new PushDownQueryDimensionSpec(ds);
+        newDims.add(dim);
+      }
+      queryWithRenamedDimensions = ((GroupByQuery) queryParam).withDimensionSpecs(newDims);
+    }
+    final GroupByQuery query = wasQueryPushedDown ? queryWithRenamedDimensions : (GroupByQuery) queryParam;
     final GroupByQueryConfig querySpecificConfig = config.withOverrides(query);
 
     final AggregatorFactory[] aggregatorFactories = new AggregatorFactory[query.getAggregatorSpecs().size()];
     for (int i = 0; i < query.getAggregatorSpecs().size(); i++) {
-      aggregatorFactories[i] = query.getAggregatorSpecs().get(i);
+      AggregatorFactory af = query.getAggregatorSpecs().get(i);
+      aggregatorFactories[i] = wasQueryPushedDown ? af.getCombiningFactory() : af;
```

Comment:
Why is the combining factory used here?

Reply:
In case of nested query push down, the historicals are going to return aggregated results and the broker's job is to combine these aggregated results - hence the use of combining aggregator factory.

Reply:
Thanks.

Can we this kind of query rewriting for dimensions and aggregatorFactories in `GroupByQueryQueryToolChect.getResultsOnPushDown()`? It would be better because we can put necessary stuffs for nestedQueryPushDown together when `isNestedQueryPushDown()` is checked.